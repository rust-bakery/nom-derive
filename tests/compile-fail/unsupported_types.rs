extern crate nom;
extern crate nom_derive;
use nom_derive::Nom;
use std::collections::HashMap;

// multiple errors, HashMap does not implement Parse
#[derive(Nom)]
pub struct S1 {
    h: HashMap<u64, u64>,
}

#[derive(Nom)]
pub struct S2 {
    h: ::std::primitive::u64, // ERROR: Nom-derive: multiple segments in type path are not supported
}

fn main() {}
