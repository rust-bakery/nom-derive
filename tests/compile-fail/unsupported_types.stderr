error: Nom-derive: multiple segments in type path are not supported
  --> $DIR/unsupported_types.rs:14:8
   |
14 |     h: ::std::primitive::u64, // ERROR: Nom-derive: multiple segments in type path are not supported
   |        ^^

error[E0599]: no function or associated item named `parse_be` found for struct `HashMap<u64, u64>` in the current scope
 --> $DIR/unsupported_types.rs:7:10
  |
7 | #[derive(Nom)]
  |          ^^^ function or associated item not found in `HashMap<u64, u64>`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no function or associated item named `parse_le` found for struct `HashMap<u64, u64>` in the current scope
 --> $DIR/unsupported_types.rs:7:10
  |
7 | #[derive(Nom)]
  |          ^^^ function or associated item not found in `HashMap<u64, u64>`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
